packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,58 @@
+import { PluginOption } from "vite";
+import upath from "upath";
+import { fileURLToPath } from "url";
+const currentDir = upath.dirname(fileURLToPath(import.meta.url));
+const packagesDir = upath.resolve(currentDir, "../..");
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        const resolvedId = (
+          await this.resolve(id, importer, { skipSelf: true })
+        )?.id;
+        if (resolvedId) {
+          return addQueryParameter(resolvedId, "inline");
+        }
+        throw new Error(
+          "Styling file could not be resolved by yext-studio-style-plugin"
+        );
+      }
+    },
+  };
+}
+
+function isImportedByStudio(importer: string) {
+  const unixImporter = upath.toUnix(importer);
+  return !!(
+    unixImporter.match(/.*\/node_modules\/@yext\/studio.*/) ||
```

Comment:
looks like this would match any of our Studio packages (`studio-plugin`, `studio-ui`, or `studio`). is that what we want?

Reply:
Yeah, I think we want any styling from any "studio" package to be ignored by the plugin since we want normal Studio styling to be injected.  We have the `tailwind-directives` from `studio` and `react-toastify` from `studio-ui`.  I guess we don't have anything from `studio-plugin` though.

Reply:
sounds good! I suppose if we ever release any other `@yext/studio-blah` repos outside of this monorepo we may want to update, but probably not important for now